File: weather_script.js
// Replace with your own API key
const API_KEY = 'your_api_key_here';
const BASE_URL = 'http://api.openweathermap.org/data/2.5/weather';

async function getWeather(city) {
    const url = `${BASE_URL}?q=${city}&appid=${API_KEY}&units=metric`;

    try {
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error('Network response was not ok');
        }
        const data = await response.json();
        const main = data.main;
        const weather = data.weather[0];

        console.log(`City: ${data.name}`);
        console.log(`Temperature: ${main.temp}°C`);
        console.log(`Weather: ${weather.description}`);
        console.log(`Humidity: ${main.humidity}%`);
        console.log(`Pressure: ${main.pressure} hPa`);
    } catch (error) {
        console.error('There has been a problem with your fetch operation:', error);
    }
}

const city = prompt("Enter city name: ");
getWeather(city);

File: weather_script.py
import requests

# Replace with your own API key
API_KEY = '4bcafd8945f202afd375949c7ef1'
BASE_URL = 'http://api.openweathermap.org/data/2.5/weather'

def get_weather(city):
    # Construct the final URL
    url = f"{BASE_URL}?q={city}&appid={API_KEY}&units=metric"
    
    # Send a GET request to the API
    response = requests.get(url)
    
    # Check if the request was successful
    if response.status_code == 200:
        data = response.json()
        main = data['main']
        weather = data['weather'][0]
        
        # Extract and print the relevant information
        print(f"City: {data['name']}")
        print(f"Temperature: {main['temp']}°C")
        print(f"Weather: {weather['description']}")
        print(f"Humidity: {main['humidity']}%")
        print(f"Pressure: {main['pressure']} hPa")
    else:
        print("Error in the HTTP request")

if __name__ == "__main__":
    city = input("Enter city name: ")
    get_weather(city)